### PR TITLE
GOA-2125_add_prtein_complexes_to_go_terms

### PR DESCRIPTION
**Pull request by rantunesboreas**

### CODE REVIEW

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java [hunk 1]
```diff
@@ -249,4 +246,53 @@ private boolean goDiscussionExists(GOTerm.NamedURL goDiscussion, Collection<Stri
...
+        assertThat(proteinComplexExists(proteinComplex2, extractedProteinComplexes), is(true));
+    }
+
+
+    @Test
+    public void extractingProteinComplexesFromEmptyProteinComplexesListReturnsNull() throws Exception {
+        when(term.getProteinComplexes()).thenReturn(null);
+
+        Optional<OntologyDocument> docOpt = converter.apply(Optional.of(term));
+        List<String> extractedProteinComplxes =
```
Comment (eddturner) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java]: typo -- really small. `...Complexes`

File: ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java [hunk 2]
```diff
@@ -75,6 +76,7 @@
         public static final String ANCESTOR = storeAndGet(VALUES, OntologyFields.ANCESTOR);
         public static final String GO_DISCUSSIONS = storeAndGet(VALUES, OntologyFields.GO_DISCUSSIONS);
         public static final String CREDITS = storeAndGet(VALUES, OntologyFields.CREDITS);
+        public final static String PROTEIN_COMPLEXS = storeAndGet(VALUES, OntologyFields.PROTEIN_COMPLEXS);
```
Comment (eddturner) [inline ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java]: `PROTEIN_COMPLEXES`

File: ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java [hunk 3]
```diff
@@ -44,6 +44,7 @@
     public final static String EXACT_SYNONYM = "exact_synonym";
     public final static String GO_DISCUSSIONS = "goDiscussions";
     public final static String CREDITS = "credits";
+    public final static String PROTEIN_COMPLEXS = "proteinComplexes";
```
Comment (eddturner) [inline ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java]: `PROTEIN_COMPLEXES`

File: ontology-rest/src/test/java/uk/ac/ebi/quickgo/ontology/controller/GOControllerIT.java [hunk 4]
```diff
@@ -55,6 +55,16 @@ public void canRetrieveGoDiscussions() throws Exception {
                 .andExpect(status().isOk());
     }
 
+    @Test
+    public void canRetrieveProteinComplexes() throws Exception {
+        ResultActions response = mockMvc.perform(get(buildTermsURLWithSubResource(GO_0000001, COMPLETE_SUB_RESOURCE)));
```
Comment (eddturner) [inline ontology-rest/src/test/java/uk/ac/ebi/quickgo/ontology/controller/GOControllerIT.java]: Just double checking, protein complexes aren't a core field right, and so should be retrieved only via the `complete` sub-resource? If so, all good. Ignore this comment.

**Comment by eddturner** [comment]
Okay -- woops. I accidentally pressed the hyperlink for "approve changes". I thought this wouldn't approve them until the next page (I'd expected a button to do the real approving!).

Anyway, could you just see to the several tiny comments I made above?

#lessonlearned :)

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java (hunk 1 above)
Comment (rantunesboreas) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java]: Done

File: ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java (hunk 3 above)
Comment (rantunesboreas) [inline ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java]: Done

File: ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java (hunk 2 above)
Comment (rantunesboreas) [inline ontology-common/src/main/java/uk/ac/ebi/quickgo/ontology/common/document/OntologyFields.java]: Done

**Pull request merged**